src/typeDefs/types.ts

```diff
@@ -35,13 +37,14 @@ export const types = gql`
   }
 
   type Comment {
-    _id: ID
+    _id: ID!
     text: String!
-    createdAt: DateTime
-    creator: User!
     post: Post!
     likedBy: [User]
     likeCount: Int
+    creator: User!
```

Comment:
make the creator field nullable

this isn't limited to this field, by making the related object fields non nullable you make your response object susceptible to error propagation that turns the whole graphql response into null instead of just turning the field that errored out to be null

Reply:
I assume, I will also have to remove `required: true` from mongoose schema, to align database schema and graphql schema

Reply:
got it @xoldyckk , thank you for the good explanation.
I kept creatorId as `required: true` in the database schema